packages/react/src/FileList/FileList.tsx

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license EUPL-1.2+
+ * Copyright Gemeente Amsterdam
+ */
+
+import { DocumentIcon } from '@amsterdam/design-system-react-icons'
+import clsx from 'clsx'
+import { forwardRef } from 'react'
+import type { ForwardedRef, HTMLAttributes } from 'react'
+import { Button } from '../Button'
+import { Icon } from '../Icon'
+import { formatFileSize } from '../common/formatFileSize'
+import { formatFileType } from '../common/formatFileType'
+
+export type FileListProps = {
+  files: FileList
+  // eslint-disable-next-line no-unused-vars
+  onDelete?: (index: number) => void
+} & HTMLAttributes<HTMLOListElement>
+
+export const FileList = forwardRef(
+  ({ files, onDelete, className, ...restProps }: FileListProps, ref: ForwardedRef<HTMLOListElement>) => (
+    <ul {...restProps} ref={ref} className={clsx('ams-file-list', className)}>
+      {Array.from(files).map((file, index) => (
```

Comment:
@alimpens and @VincentSmedinga if I understand correctly, you want user to build this loop themselves?

Reply:
Yes, please. Subcomponents offers them more control, and it relieves us from the object prop.